### PR TITLE
Customize media notifications through Media Session API

### LINKED ISSUE #11: Play nice with Chrome 79's new Media Playback Controls
Chrome 79 brought a new [Media Playback Control feature](https://beebom.com/chrome-79-media-playback-controls-address-bar/) that is still rolling out to some users. It appeared on my browser yesterday.

This is how it looks:
Screenshot 2020-01-12 at 12 59 20

It is basically a button in the sidebar that shows up when something in the browser is playing music. 

I guess its intended use is to better control music playing on other tabs e.g youtube. But, as you can see above, it will also trigger whenever the extension plays music. And the result is not nice.

We could investigate this new feature (did some quick research and did not found any official documentation) and either disable it for our extension (if that's even possible) or leverage it and try to use the artist, title and album art fields to our advantage. 

We might even enable 'back' and 'next' buttons when listening to KK!

### PR DESCRIPTION
**Pull request by jordigarcl**
This PR would close #11 

I have implemented a simple `MediaSessionManager` to define the MediaSession metadata that is shown on Chrome's media notification.

**Changes**
- Metadata is shown when playing hourly music

**TODO**
- KK Integration

### CODE REVIEW

File: manifest.json [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+         "scripts/LoopTimes.js", 
+         "scripts/WeatherManager.js", 
+         "scripts/KKSongs.js",
+         "scripts/AudioManager.js", 
+         "scripts/BadgeManager.js", 
+         "scripts/NotificationManager.js", 
+         "scripts/StateManager.js", 
+         "scripts/MediaSessionManager.js", 
+         "scripts/main.js" 
+      ]
```
Comment (jordigarcl) [inline manifest.json:17]: Given the current `scripts` size I believe this provides better visibility and makes it easier to add new entries

File: scripts/MediaSessionManager.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	let gameNames = {
+		'animal-crossing': 'Animal Crossing',
+		'wild-world': 'Animal Crossing: Wild World',
+		'new-leaf': 'Animal Crossing: New Leaf',
+	}
+	let gameArtwork = {
+		'animal-crossing': 'https://i.imgur.com/dgxbXpm.png',
+		'wild-world': 'https://i.imgur.com/U9MW9J7.png',
+		'new-leaf': 'https://i.imgur.com/nfpxWtV.png'
+	}
```
Comment (jordigarcl) [inline scripts/MediaSessionManager.js]: What do you guys think about these covers? I aimed to leave out all labels from the original covers while still featuring the Animal Crossing logos.

File: manifest.json (hunk 1 above)
Comment (PikaDude) [reply manifest.json:17]: Looks much better.

File: scripts/MediaSessionManager.js (hunk 2 above)
Comment (PikaDude) [reply scripts/MediaSessionManager.js]: They look good! BTW, we probably shouldn't read them from Imgur, we should read them locally.

Comment (jordigarcl) [reply scripts/MediaSessionManager.js]: I originally tried setting a relative path to the image, but it didn't work. Chrome complained:
```
MediaImage src can only be of http/https/data/blob scheme
```
As can be seen [here](https://code.immersaview.com/remotes/chromium-third_party/blob/d2bb3bb52ed2e529511b71b85331181ec7fb5464/WebKit/Source/modules/mediasession/MediaMetadataSanitizer.cpp) it looks like Chrome implementation of MediaSessionAPI is expecting a URL in that field. 

More examples here:
https://googlechrome.github.io/samples/media-session/audio.html

Comment (PikaDude) [reply scripts/MediaSessionManager.js]: Well, that's kind of dumb. We can use the data scheme and encode the albums in Base64. I think we should try and keep them offline personally.

Comment (jordigarcl) [reply scripts/MediaSessionManager.js]: Great suggestion. I just tried it and it loads base64 artwork flawlessly. Will push changes soon! 👍

Comment (PikaDude) [reply scripts/MediaSessionManager.js]: Already way ahead of you. I'll push everything when I'm finished.

**Review by PikaDude — COMMENTED** [review]
The plan is to store KK's album art locally. All 75 album artwork will be bundled with the extension upon release, but will not be stored on GitHub. It will be available in a Google Drive link for GitHub users linked in the README. As a fallback, it will grab a copy of album art from my server - for GitHub users who have not downloaded the resources.

**Review by PikaDude — APPROVED** [review]

**Pull request merged**